app/models/required_address_validator.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Validates that either an address_state or a foreign_address_country is present
+class RequiredAddressValidator < ActiveModel::Validator
+  def validate(record)
+    record.errors.add(:base, "One of 'State' or 'Foreign Address Country' is required") if record.address_state.blank? && record.foreign_address_country.blank?
```

Comment:
Shouldn't this be updated to match the front end validation? i.e. address line 1/city/state/zip are required for domestic address and city/country required for foreign address

Reply:
This was actually an intentional choice to be less restrictive in the API. See this PR here: https://github.com/SaraAlert/SaraAlert/pull/501. And we came to this decision here: https://mitrecorp.slack.com/team/WM4DX564V.

Reply:
Oops forgot about that. What a throwback!